src/store/modules/ui.store.js

```diff
@@ -412,13 +421,22 @@ export default {
                 commit('removeError', { error, dispatcher })
             }
         },
-        addWarnings({ commit }, { warnings, dispatcher }) {
+        addWarnings({ commit, state }, { warnings, dispatcher }) {
             if (
                 warnings instanceof Array &&
                 warnings.filter((warning) => warning instanceof WarningMessage).length ===
                     warnings.length
             ) {
-                commit('addWarnings', { warnings, dispatcher })
+                warnings = warnings.filter(
+                    (warning) =>
+                        // we only add the warnings that are not existing within the store
+                        [...state.warnings].filter((otherWarning) => {
+                            warning.isEquals(otherWarning)
+                        }).length === 0
+                )
```

Comment:
i'm not sure that the double filter works, i think this might be better for the warnings and the errors

```suggestion
        warnings = warnings.filter(
            (warning) =>
                // we only add the warnings that are not existing within the store
                !state.warnings.some((otherWarning) => warning.isEquals(otherWarning))
        )
```

Reply:
The `.some` method will encounter the same issue as the filter, but it will be better and easier to read. I'll have to implement the check for the `params` in warnings and errors to try to get rid of this. Thanks for the feedback :)